daemon/network.go

```diff
@@ -137,18 +142,22 @@ func (daemon *Daemon) enqueueIngressJob(job *ingressJob) {
 }
 
 // SetupIngress setups ingress networking.
-func (daemon *Daemon) SetupIngress(create clustertypes.NetworkCreateRequest, nodeIP string) error {
+// If caller wants to be notified when the job is completed,
+// it needs to pass a done channel.
+func (daemon *Daemon) SetupIngress(create clustertypes.NetworkCreateRequest, nodeIP string, done chan struct{}) error {
 	ip, _, err := net.ParseCIDR(nodeIP)
 	if err != nil {
 		return err
 	}
-	daemon.enqueueIngressJob(&ingressJob{&create, ip})
+	daemon.enqueueIngressJob(&ingressJob{&create, ip, done})
 	return nil
 }
 
 // ReleaseIngress releases the ingress networking.
-func (daemon *Daemon) ReleaseIngress() error {
-	daemon.enqueueIngressJob(&ingressJob{nil, nil})
+// If caller wants to be notified when the job is completed,
+// it needs to pass a done channel.
+func (daemon *Daemon) ReleaseIngress(done chan struct{}) error {
```

Comment:
Similarly to my comment on `SetupIngress`, `ReleaseIngress` could return a channel that gets closed when the job is completed.

```go
func (daemon *Daemon) ReleaseIngress() (<-chan struct, error) {
```